src/EFCore.PG/Query/Sql/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -176,6 +192,15 @@ public Expression VisitArrayAny(ArrayAnyExpression arrayAnyExpression)
             return arrayAnyExpression;
         }
 
+        public Expression VisitDictionaryContainsKey(DictionaryContainsKeyExpression expr)
+        {
+            Visit(expr.Dictionary);
```

Comment:
This could be written using SqlFunctionExpression in 2.1 passing in dictionary as instance. That would avoid creating new expression type